internal/specs/specs.go

```diff
@@ -361,3 +361,31 @@ func LoadFromCluster(ctx context.Context, client kubernetes.Interface, selectors
 		RawSpecs: rawSpecs,
 	})
 }
+
+// LoadAdditionalSpecFromURIs loads additional specs from the URIs provided in the troubleshoot kinds.
+// This function will modify kinds in place.
+func LoadAdditionalSpecFromURIs(ctx context.Context, kinds *loader.TroubleshootKinds) {
+	uris := kinds.GetURIs()
```

Comment:
At this point, you loose context of what uri belongs to what spec

Reply:
Thanks @banjoh! May I know why do we have to maintain the context of URI-spec? As the final result is a merged object from all sources (argument, secret, config map, etc.)

Reply:
A spec downloaded from a `uri` is meant to[replace the original one entirely, not merge](https://github.com/replicatedhq/troubleshoot/blob/main/docs/design/proposal-updateable-specs.md#goals). Without knowing which spec the url is linked to, it won't be possible to know which spec needs to be replaced.

Reply:
Argh I missed this design doc, let me update the code accordingly.